extension/intents/slideshow/contentScript.js

```diff
@@ -0,0 +1,186 @@
+/* globals communicate */
+
+this.slideshowScript = (function() {
+  let slideElements = [];
+  let currentSlideIndex = 0;
+
+  const videoSources = ["youtube.com", "vimeo.com"];
+
+  slideElements = detectAllMedia();
+  buildSlideStructure();
+
+  const lbContainer = document.body.querySelector(".fv-lightbox-container");
+  const slideContainer = document.body.querySelector(".fv-slide-image");
+
+  window.onresize = function() {
+    if (slideContainer.hasChildNodes()) {
+      if (slideContainer.firstChild.tagName === "IFRAME") {
+        slideContainer.firstChild.setAttribute(
+          "width",
+          document.documentElement.clientWidth
+        );
+        slideContainer.firstChild.setAttribute(
+          "height",
+          document.documentElement.clientHeight
+        );
+      }
+    }
+  };
+
+  function buildSlideStructure() {
+    const lightboxElement = document.createElement("div");
```

Comment:
There's a danger this could conflict with the underlying page, but that's something that could be fixed in a followup.

But to give a hint, how this would generally be fixed would be something like:

```js
const iframe = document.createElement("iframe");
iframe.src = browser.runtime.getURL("/intents/slideshow/slideshow.html");
iframe.style.position = "absolute"; // and some other positioning styles
document.body.appendChild(iframe);
```

And you can manipulate the contents of the iframe much like you would the original document, just using `iframe.contentDocument` instead of `document`.

`slideshow.html` could also have its own stylesheet referenced from the file itself.

Reply:
Thanks @ianb for your recommendations. By conflicts, I'm guessing you mean stylesheets from various sites could end up disrupting the slideshow's structure? I didn't know I could reference and use `html` files in this way, would've made things easier to visualize, easier to implement. I'll get to work on these right away

Reply:
@ianb I have modified the slideshow to use an iframe as you advised. However, this has brought on a challenge. At the point where I try to append the new Iframe object with given src to the document object, I get this security error "Security Error: Content at http://xxxx.xxx/xxxx may not load or link to moz-extension://a669dc66-0200-48e7-83da-e7a815fac9f5/intents/slideshow/slideshow.html". From observation this seems to happen when the Iframe's src is being parsed. I felt this was a permissions issue. I tried to add the slideshow.html to "web_accessible_resources" in manifest.json. Still no changes. Please, I need your help in moving forward from this point

Reply:
@ianb Finally got the slideshow into an iframe (TIL Iframe's are a pain). Got past cross domain issue/permissions by skipping src altogether. And found out much hair-pulling that contentDocument is only accessible after "onload" event. Thanks for these suggestions I have learned a lot